### ISSUE TITLE
Regression on ValidatorServiceProvider

### ISSUE THREAD

**Issue by olivier34000** [silexphp/Silex#1213]
I updated the last version of Silex, and there is a regression with ValidatorServiceProvider on the master branch. The following test is broken on master and it works on 1.3.
```
<?php

namespace Silex\Tests\Provider;

use Silex\Application;
use Silex\Provider\TranslationServiceProvider;
use Silex\Provider\ValidatorServiceProvider;

class ValidatorServiceProvider2Test extends \PHPUnit_Framework_TestCase
{
    public function testAddResourceAlternate2()
    {
        $app = new Application();
        $app['locale'] = 'fr';
        unset($app['exception_handler']);
        $app->register(new TranslationServiceProvider());
        $app->register(new ValidatorServiceProvider());
        $app->get('/test', function () use ($app) {
            $test = $app['validator'];
            return new \Symfony\Component\HttpFoundation\Response('');
        });

        $client = new \Symfony\Component\HttpKernel\Client($app);
        $client->request('GET', '/test');
        $response = $client->getResponse();
        $this->assertEquals(200, $response->getStatusCode());
    }
}
```

**Comment by stof** [comment]
Well, please provide more detauls about what the regression is. What is the error ?

**Comment by olivier34000** [comment]
The error 
```
There was 1 error:

1) Silex\Tests\Provider\ValidatorServiceProvider2Test::testAddResourceAlternate2
InvalidArgumentException: Identifier "translator.resources" does not contain an object definition.
```

When the application is executed TranslatorListener is called in TranslationServiceProvider.  $app['translator'] is executed and we loop on $app['translator.resources']. 
However when we call $app['validator'] we extend $app['translator.resources'] and Pimple does not allow it.

**Comment by fabpot** [comment]
On master, we should wrap the transltor.resources definition with a `protect` call. Can you try and submit a PR if that works?

**Comment by patrick-fls** [comment]
I can confirm it works to wrap it with `protect`

**Comment by fabpot** [comment]
fixed in 48a3fdc
